### ISSUE TITLE
Powershell Hashtable Keys are case-sensitive on Ubuntu 18.04

### ISSUE THREAD

**Issue by dantraMSFT** [PowerShell/PowerShell#7761]
On a clean installation of Ubuntu 18.04 from Azure, creating a hashtable in script results in a hashtable with case-sensitive keys.

The LANG environment variable is set to 'C.UTF-8' and both CultureInfo.CurrentCulture and CultureInfo.CurrentUICulture report  en-US-POSIX.

The result is scripts that are not case-strict when using hashtable keys will break.

Steps to reproduce
------------------
```powershell
$h = @{HostName = 'host'}
$h.hostname
$h.HostName
```

Expected behavior
-----------------
The value 'host' is returned for both $h.HostName and $h.hostname

Actual behavior
---------------
No value is returned for $h.HostName is 'host' while $h.hostname returns $null.

Environment data
----------------

```powershell
> $PSVersionTable
 Name                           Value
----                           -----
PSVersion                      6.1.0-rc.1
PSEdition                      Core
GitCommitId                    6.1.0-rc.1
OS                             Linux 4.15.0-1022-azure #23-Ubuntu SMP Thu Aug 16 15:19:42 UTC 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by BrucePay** [comment]
How odd. The code that constructs a hashtable is hard-coded to be case-insensitive:
```c#
            yield return Expression.Assign(temp,
                Expression.New(ordered ? CachedReflectionInfo.OrderedDictionary_ctor : CachedReflectionInfo.Hashtable_ctor,
                                ExpressionCache.Constant(keyValuePairs.Count),
                                ExpressionCache.CurrentCultureIgnoreCaseComparer.Cast(typeof(IEqualityComparer))));
```
What happens when you manually create the hashtable?
```powershell
$h = [hashtable]::new([StringComparer]::CurrentCultureIgnoreCase)
$h.hostname = 123
$h.HostName
$h.hostname
```

**Comment by rjmholt** [comment]
@JamesWTruher found that this is due to Ubuntu 18.04's default [`LANG=C.UTF-8`](https://manpages.debian.org/unstable/open-infrastructure-locales-c.utf-8/locales-c.utf-8.7.en.html) (compare to 16.04's `LANG=en-us.UTF-8` for our installs).

@TravisEz13 made the point that this behaviour may be by design: the if the `C.UTF-8` locale is supposed to be something along the lines of "culture invariant", then case-sensitivity may not mean anything and it just performs a strict codepoint-to-codepoint comparison.

However, from the Debian manpage linked above, it sounds like the standard C locale should obey the casing rules of en_US, meaning C.UTF-8 should obey the casing rules of en_US.UTF-8. So this is a bug, probably in .NET Core.

**Comment by rjmholt** [comment]
I've managed a simple repo that doesn't depend on PowerShell and have opened an issue in corefx

**Comment by rjmholt** [comment]
CoreFX have designated the behaviour with that locale as by design as @TravisEz13 suggested, see https://github.com/dotnet/corefx/issues/32250#issuecomment-420749205.

The options I see for our handling of it are:
- This behaviour is by design for us too
- We special-case handling of this locale
- We change handling of hashtable keys to use `CultureInvariantIgnoreCase` or `OrdinalIgnoreCase` as I think @BrucePay suggested

**Comment by rjmholt** [comment]
Apparently there are issues with `CultureInfo.InvariantCulture`: https://github.com/dotnet/roslyn/issues/29938#issuecomment-428270244

**Comment by bergmeister** [comment]
I suggest to use OrdinalIgnoreCase for the hashtable comparison please. It would be great if a fix for this gets shipped in the next patch of 6.1.
